cmd/clusterctl/pkg/client/config.go

```diff
@@ -47,19 +48,150 @@ func (c *clusterctlClient) GetProviderComponents(provider, targetNameSpace, watc
 	return components, nil
 }
 
+// GetClusterTemplateOptions carries the options supported by GetClusterTemplate.
+type GetClusterTemplateOptions struct {
+	// Kubeconfig file to use for accessing the management cluster. If empty, default rules for kubeconfig
+	// discovery will be used.
+	Kubeconfig string
+
+	// ProviderRepositorySource to be used for reading the workload cluster template from a provider repository;
+	// only one template source can be used at time; if not other source will be set, a ProviderRepositorySource
+	// will be generated inferring values from the cluster.
+	ProviderRepositorySource *ProviderRepositorySourceOptions
+
+	// URLSource to be used for reading the workload cluster template; only one template source can be used at time.
+	URLSource *URLSourceOptions
+
+	// ConfigMapSource to be used for reading the workload cluster template; only one template source can be used at time.
+	ConfigMapSource *ConfigMapSourceOptions
+
+	// TargetNamespace where the objects describing the workload cluster should be deployed. If not specified,
+	// the current namespace will be used.
+	TargetNamespace string
+
+	// ClusterName to be used for the workload cluster.
+	ClusterName string
+
+	// KubernetesVersion to use for the workload cluster. By default (empty), the value from os env variables
+	// or the .cluster-api/clusterctl.yaml config file will be used.
+	KubernetesVersion string
+
+	// ControlPlaneMachineCount defines the number of control plane machines to be added to the workload cluster.
+	ControlPlaneMachineCount int
+
+	// WorkerMachineCount defines number of worker machines to be added to the workload cluster.
+	WorkerMachineCount int
+
+	// listVariablesOnly sets the GetClusterTemplate method to return the list of variables expected by the template
+	// without executing any further processing.
+	ListVariablesOnly bool
+}
+
+// numSources return the number of template sources currently set on a GetClusterTemplateOptions.
+func (o *GetClusterTemplateOptions) numSources() int {
```

Comment:
This method seems to be used to check if the source is valid, seems that having more than one source isn't something we want to support. For that reason I'd suggest renaming to `validateSources() error`

Reply:
we need to know the number of sources
- if > 1 error
- if == 0 use Respostiotry as a default source (values will be inferred from the cluster inventory)